core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -0,0 +1,489 @@
+  private final Map<UUID, PlayerStats> allPlayerStats = new HashMap<>();
+  // Since Bukkit#getOfflinePlayer reads the cached user files, and those files have an expire date
+  // + will be wiped if X amount of players join, we need a separate cache for players with stats
+  private final Map<UUID, String> cachedUsernames = new HashMap<>();
+
+  private final boolean verboseStats = PGM.get().getConfiguration().showVerboseStats();
+  private final Component verboseStatsTitle = TranslatableComponent.of("match.stats.title");
+
+  // Defined at match end, see #onMatchEnd
+  InventoryMenu endOfMatchMenu;
+
+  public StatsMatchModule(Match match) {
+    this.match = match;
+  }
+
+  public static class PlayerStats {
+    // K/D
+    int kills;
+    int deaths;
+    int killstreak;
+    int killstreakMax;
+
+    // Bow
+    int longestBowKill;
+    private double bowDamage;
+    int shotsTaken;
+    int shotsHit;
+
+    // Damage
+    private double damageDone;
+    private double damageTaken;
+
+    // Objectives
+    int destroyablePiecesBroken;
+    int flagsCaptured;
+
+    Duration longestFlagHold = Duration.ZERO;
+    private Instant longestFlagHoldCache;
+
+    // The task responsible for displaying the stats over the hotbar
+    // See #sendLongHotbarMessage
+    private Future<?> hotbarTask;
+
+    private void onMurder() {
+      kills++;
+      killstreak++;
+      if (killstreak > killstreakMax) killstreakMax = killstreak;
+    }
+
+    private void onDeath() {
+      deaths++;
+      killstreak = 0;
+    }
+
+    private void onDamage(double damage, boolean bow) {
+      damageDone += damage;
+      if (bow) {
+        bowDamage += damage;
+        shotsHit++;
+      }
+    }
+
+    private void onDamaged(double damage) {
+      damageTaken += damage;
+    }
+
+    private void onDestroyablePieceBroken(int change) {
+      destroyablePiecesBroken += change;
+    }
+
+    private void onFlagCapture() {
+      flagsCaptured++;
+      onFlagDrop();
+    }
+
+    private void onFlagPickup() {
+      longestFlagHoldCache = Instant.now();
+    }
+
+    private void onFlagDrop() {
+      setLongestFlagHold(
+          Duration.ofMillis(Instant.now().toEpochMilli() - longestFlagHoldCache.toEpochMilli()));
+    }
+
+    private void setLongestFlagHold(Duration time) {
+      if (longestFlagHold == null || (time.toNanos() - longestFlagHold.toNanos()) > 0)
+        longestFlagHold = time;
+    }
+
+    private void setLongestBowKill(double distance) {
+      if (distance > longestBowKill) {
+        longestBowKill = (int) Math.ceil(distance);
+      }
+    }
+
+    private void onBowShoot() {
+      shotsTaken++;
+    }
+
+    /** A common format used by all stats with decimals */
+    public static final DecimalFormat STATS_DECIMALFORMAT = new DecimalFormat("#.##");
+
+    public static double statFormat(Number number) {
+      String stat = STATS_DECIMALFORMAT.format(number).replace(',', '.');
+      return Double.parseDouble(stat);
+    }
```

Comment:
Don't convert a double into a double. Format when you're about to display it to the user, when creating the TextComponent. Here's you're not setting format, you're just arbitrarily altering doubles to try to force a specific rounding